src/Form/FormAnalyzer.js

```diff
@@ -49,20 +56,37 @@ class FormAnalyzer {
         return this
     }
 
+    /**
+     *
+     * @param {object} p
+     * @param {'positive'|'negative' | ''} [p.matches] - Skip regexes and just update the value
+     * @param {string} p.string - The string to check
+     * @param {number} p.strength - Strength of the signal
+     * @param {string} [p.signalType] - For debugging purposes, we give a name to the signal
+     * @param {boolean} [p.shouldFlip] - Flips the signals, i.e. when a link points outside. See below
+     * @param {boolean} [p.shouldCheckUnifiedForm] - Should check for login/signup forms
+     * @param {boolean} [p.shouldBeConservative] - Should use the conservative signup regex
+     * @returns {FormAnalyzer}
+     */
```

Comment:
I've added these as types and added the new parameter `matches`. It allows us to skip the regex checks in some cases when we the signal is not directly related to a string.

Reply:
❓ Out of interest, in these cases why not just call `increaseSignalBy` directly instead of calling it through `updateSignal`?

Reply:
Yeah, my original idea was to have only one way to update the signal and it was through `updateSignal`, but this is just adding unnecessary complexity, so you're right. I will push a change.